Uplift/Views/WeeklyWorkoutTrackerView.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  WeeklyWorkoutTrackerView.swift
+//  Uplift
+//
+//  Created by jiwon jeong on 3/9/25.
+//  Copyright © 2025 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+
+struct WeeklyWorkoutTrackerView: View {
+    @ObservedObject var viewModel: ProfileView.ViewModel
+    @State private var animationProgress: Double = 0
+
+    // Weekday abbreviations
+    private let weekdays = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
+
+    // Days that have workouts
+    @State private var workoutDays: [Bool] = [false, false, false, false, false, false, false]
+
+    // Animation timing
+    private let animationDuration: Double = 0.05
+    private let delayBetweenDays: Double = 0.2
+
+    // Circle dimensions
+    private let circleSize: CGFloat = 24
+    private let lineWidth: CGFloat = 2
+    private let spacing: CGFloat = 26.5
+    private let verticalSpacing: CGFloat = 2
+
+    var body: some View {
+        VStack {
+            VStack(alignment: .center, spacing: verticalSpacing) {
+                // Weekday abbreviations
+                HStack(spacing: spacing) {
+                    ForEach(0..<7, id: \.self) { index in
```

Comment:
This magic number is technically since the number of days of the week prob will not change anytime soon lol, but just for good practice, I would do `weekdays.indices`